src/network/tc/qdisc.c

```diff
@@ -132,25 +133,29 @@ int qdisc_configure(Link *link, QDiscs *qdisc) {
                 return log_link_error_errno(link, r, "Could not create tcm_parent message: %m");
 
         if (qdisc->parent == TC_H_CLSACT) {
+                tca_kind = strdup("clsact");
+                if (!tca_kind)
+                        return log_oom();
+
                 r = sd_rtnl_message_set_qdisc_handle(req, TC_H_MAKE(TC_H_CLSACT, 0));
                 if (r < 0)
                         return log_link_error_errno(link, r, "Could not set tcm_handle message: %m");
-
-                r = sd_netlink_message_append_string(req, TCA_KIND, "clsact");
-                if (r < 0)
-                        return log_link_error_errno(link, r, "Could not append TCA_KIND attribute: %m");
         }
 
         if (qdisc->has_network_emulator) {
-                r = sd_netlink_message_append_string(req, TCA_KIND, "netem");
-                if (r < 0)
-                        return log_link_error_errno(link, r, "Could not append TCA_KIND attribute: %m");
+                tca_kind = strdup("netem");
```

Comment:
So, if `NetworkEmulatorLossRate=` is set, then `Parent=clsact` is partially ignored?

At least, this needs to be freed before `strdup()`, so, please use `free_and_strdup()`.

Reply:
We can add `clsact` like `sudo tc qdisc add dev veth99 clsact`
here no Netem conf is there but still then we able to create . By doing this we can replicate the same.  If Netem conf is available then we have to take that in consideration and apply Netem conf with clsact.

```
[TrafficControlQueueingDiscipline]
Parent=clsact
```